### PR TITLE
Ftr:  add the module of shopping-order

### PR DESCRIPTION
**Pull request by PhilYue**
1. add the module of shopping-order, contains examples of distributed solutions by Seata
2. add the config of .run
3. update dubbo-go version,  v1.5.6-rc1 to v1.5.6-rc2

### CODE REVIEW

File: shopping-order/go-client/cmd/client.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/apache/dubbo-go-samples/shopping-order/go-client/pkg"
```
Comment (AlexStocks) [inline shopping-order/go-client/cmd/client.go:23]: move it to the 3rd import block.

File: shopping-order/go-client/conf/client.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      tcp_keep_alive: true
+      keep_alive_period: "120s"
+      tcp_r_buf_size: 262144
+      tcp_w_buf_size: 65536
+      pkg_rq_size: 1024
+      pkg_wq_size: 512
+      tcp_read_timeout: "1s"
+      tcp_write_timeout: "5s"
+      wait_timeout: "1s"
+      max_msg_len: 10240
```
Comment (AlexStocks) [inline shopping-order/go-client/conf/client.yml:62]: change it to 8388608 which is  8MiB

File: shopping-order/go-client/conf/seata.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    compress_encoding: false
+    tcp_no_delay: true
+    tcp_keep_alive: true
+    keep_alive_period: "180s"
+    tcp_r_buf_size: 262144
+    tcp_w_buf_size: 65536
+    tcp_read_timeout: "1s"
+    tcp_write_timeout: "5s"
+    wait_timeout: "1s"
+    max_msg_len: 4096
```
Comment (AlexStocks) [inline shopping-order/go-client/conf/seata.yml:19]: change it to 8388608

File: shopping-order/go-server-common/filter/seata_filter.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package filter
+
+import (
+	"context"
+)
+import (
```
Comment (AlexStocks) [inline shopping-order/go-server-common/filter/seata_filter.go:6]: add blank above.

File: shopping-order/go-server-common/filter/seata_filter.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/filter"
+	"github.com/apache/dubbo-go/protocol"
+)
+import (
```
Comment (AlexStocks) [inline shopping-order/go-server-common/filter/seata_filter.go:11]: delete this blank import block.

File: shopping-order/go-server-common/filter/seata_filter.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+func init() {
+	extension.SetFilter(SEATA, getSeataFilter)
+}
+
+// SeataFilter ...
+type SeataFilter struct {
+}
+
+// Invoke ...
```
Comment (AlexStocks) [inline shopping-order/go-server-common/filter/seata_filter.go:28]: add comment

File: shopping-order/go-server-common/filter/seata_filter.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+const (
+	SEATA     = "seata"
+	SEATA_XID = "seata_xid"
+)
+
+func init() {
+	extension.SetFilter(SEATA, getSeataFilter)
+}
+
+// SeataFilter ...
```
Comment (AlexStocks) [inline shopping-order/go-server-common/filter/seata_filter.go:24]: // nolint

File: shopping-order/go-server-common/filter/seata_filter.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+// Invoke ...
+func (sf *SeataFilter) Invoke(ctx context.Context, invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	xid := invocation.AttachmentsByKey(SEATA_XID, "")
+	if xid != "" {
+		return invoker.Invoke(context.WithValue(ctx, SEATA_XID, xid), invocation)
+	}
+	return invoker.Invoke(ctx, invocation)
+}
+
+// OnResponse ...
```
Comment (AlexStocks) [inline shopping-order/go-server-common/filter/seata_filter.go:37]: add comment

File: shopping-order/go-server-common/filter/seata_filter.go [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+func (sf *SeataFilter) OnResponse(ctx context.Context, result protocol.Result, invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	return result
+}
+
+// GetSeataFilter ...
```
Comment (AlexStocks) [inline shopping-order/go-server-common/filter/seata_filter.go:42]: delete this comment

File: shopping-order/go-server-order/cmd/server.go [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+	_ "github.com/apache/dubbo-go/metadata/service/inmemory"
+	_ "github.com/apache/dubbo-go/protocol/dubbo"
+	_ "github.com/apache/dubbo-go/registry/protocol"
+	"github.com/transaction-wg/seata-golang/pkg/client"
+	"github.com/transaction-wg/seata-golang/pkg/client/at/exec"
+	seataConfig "github.com/transaction-wg/seata-golang/pkg/client/config"
+	//_ "github.com/apache/dubbo-go/registry/nacos"
+	_ "github.com/apache/dubbo-go/registry/zookeeper"
+)
+import (
```
Comment (AlexStocks) [inline shopping-order/go-server-order/cmd/server.go:44]: add blank line above

File: shopping-order/go-server-order/conf/seata.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    compress_encoding: false
+    tcp_no_delay: true
+    tcp_keep_alive: true
+    keep_alive_period: "180s"
+    tcp_r_buf_size: 262144
+    tcp_w_buf_size: 65536
+    tcp_read_timeout: "1s"
+    tcp_write_timeout: "5s"
+    wait_timeout: "1s"
+    max_msg_len: 4096
```
Comment (AlexStocks) [inline shopping-order/go-server-order/conf/seata.yml:18]: change it to 8688608

File: shopping-order/go-server-order/conf/server.yml [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+      tcp_keep_alive: true
+      keep_alive_period: "120s"
+      tcp_r_buf_size: 262144
+      tcp_w_buf_size: 65536
+      pkg_rq_size: 1024
+      pkg_wq_size: 512
+      tcp_read_timeout: "1s"
+      tcp_write_timeout: "5s"
+      wait_timeout: "1s"
+      max_msg_len: 1024
```
Comment (AlexStocks) [inline shopping-order/go-server-order/conf/server.yml:50]: change it to 8388608

File: shopping-order/go-server-product/cmd/server.go [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
```
Comment (AlexStocks) [inline shopping-order/go-server-product/cmd/server.go:27]: pls add blank line btw import block.

File: shopping-order/go-server-product/cmd/server.go [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+	_ "github.com/apache/dubbo-go/metadata/service/inmemory"
+	_ "github.com/apache/dubbo-go/protocol/dubbo"
+	_ "github.com/apache/dubbo-go/registry/protocol"
+	"github.com/transaction-wg/seata-golang/pkg/client"
+	"github.com/transaction-wg/seata-golang/pkg/client/at/exec"
+	seataConfig "github.com/transaction-wg/seata-golang/pkg/client/config"
+	//_ "github.com/apache/dubbo-go/registry/nacos"
+	_ "github.com/apache/dubbo-go/registry/zookeeper"
+)
+import (
```
Comment (AlexStocks) [inline shopping-order/go-server-product/cmd/server.go:44]: add blank line above

File: shopping-order/go-server-product/conf/seata.yml [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    compress_encoding: false
+    tcp_no_delay: true
+    tcp_keep_alive: true
+    keep_alive_period: "180s"
+    tcp_r_buf_size: 262144
+    tcp_w_buf_size: 65536
+    tcp_read_timeout: "1s"
+    tcp_write_timeout: "5s"
+    wait_timeout: "1s"
+    max_msg_len: 4096
```
Comment (AlexStocks) [inline shopping-order/go-server-product/conf/seata.yml:20]: change it to 8388608

File: shopping-order/go-server-product/conf/server.yml [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+      tcp_keep_alive: true
+      keep_alive_period: "120s"
+      tcp_r_buf_size: 262144
+      tcp_w_buf_size: 65536
+      pkg_rq_size: 1024
+      pkg_wq_size: 512
+      tcp_read_timeout: "1s"
+      tcp_write_timeout: "5s"
+      wait_timeout: "1s"
+      max_msg_len: 1024
```
Comment (AlexStocks) [inline shopping-order/go-server-product/conf/server.yml:54]: change it to 8388608

File: shopping-order/go.mod [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+	github.com/linode/linodego v0.10.0 // indirect
+	github.com/miekg/dns v1.1.27 // indirect
+	github.com/mitchellh/hashstructure v1.0.0 // indirect
+	github.com/opentrx/mysql v1.0.0-pre
+	github.com/transaction-wg/seata-golang v0.2.1-alpha
+	github.com/uber/jaeger-lib v2.2.0+incompatible // indirect
+	google.golang.org/grpc/examples v0.0.0-20210322221411-d26af8e39165 // indirect
+)
+
+replace github.com/envoyproxy/go-control-plane => github.com/envoyproxy/go-control-plane v0.8.0
```
Comment (AlexStocks) [inline shopping-order/go.mod:22]: merge the two `replace` lines.

**Review by AlexStocks — APPROVED** [review]

**Review by zhaoyunxing92 — APPROVED** [review]
approved these changes

**Review by beiwei30 — APPROVED** [review]
LGTM

**Pull request merged**